src/checkers/chromiumchecker.py

```diff
@@ -0,0 +1,186 @@
+            result = await response.text()
+
+        update_py = base64.b64decode(result).decode("utf-8")
+
+        revision_match = self._CLANG_REVISION_RE.search(update_py)
+        assert revision_match is not None, url
+
+        sub_revision_match = self._CLANG_SUB_REVISION_RE.search(update_py)
+        assert sub_revision_match is not None, url
+
+        return LLVMComponent.Version(
+            revision_match.group(1), sub_revision_match.group(1)
+        )
+
+
+class LLVMGitComponent(LLVMComponent):
+    NAME = "llvm-git"
+    DATA_CLASS = ExternalGitRepo
+
+    _LLVM_REPO_URL = "https://github.com/llvm/llvm-project"
+
+    async def check(self) -> None:
+        assert isinstance(self.external_data, ExternalGitRepo)
+
+        llvm_version = await self.get_llvm_version()
+
+        new_version = await ExternalGitRef(
+            url=self.external_data.current_version.url,
+            commit=llvm_version.revision,
+            tag=None,
+            branch=None,
+            version=self.latest_version,
+            timestamp=None,
+        ).fetch_remote()
+        self.external_data.set_new_version(new_version)
+
+
+class LLVMPrebuiltComponent(LLVMComponent):
+    NAME = "llvm-prebuilt"
+    DATA_CLASS = ExternalDataSource
+
+    _PREBUILT_URL_FORMAT = (
+        "https://commondatastorage.googleapis.com"
+        "/chromium-browser-clang/Linux_x64/clang-{revision}-{sub_revision}.tgz"
+    )
+
+    async def check(self) -> None:
+        assert isinstance(self.external_data, ExternalDataSource)
+
+        llvm_version = await self.get_llvm_version()
+
+        latest_url = self._PREBUILT_URL_FORMAT.format(
+            revision=llvm_version.revision, sub_revision=llvm_version.sub_revision
+        )
+        await self.update_external_source_version(latest_url)
+
+
+class ChromiumChecker(Checker):
+    CHECKER_DATA_TYPE = "chromium"
+    SUPPORTED_DATA_CLASSES = [ExternalData, ExternalGitRepo]
+
+    _COMPONENTS = {
+        c.NAME: c for c in (ChromiumComponent, LLVMGitComponent, LLVMPrebuiltComponent)
+    }
+
+    _CHROMIUM_VERSIONS_URL = "https://omahaproxy.appspot.com/all.json"
+    _CHROMIUM_VERSIONS_PARAMS = {"os": "linux", "channel": "stable"}
+
+    async def _get_latest_chromium(self) -> str:
+        async with self.session.get(
+            self._CHROMIUM_VERSIONS_URL, params=self._CHROMIUM_VERSIONS_PARAMS
+        ) as response:
+            result = await response.json()
+
+        assert len(result) == 1, result
+        assert len(result[0]["versions"]) == 1, result
+        return result[0]["versions"][0]["current_version"]
+
+    async def check(self, external_data: t.Union[ExternalData, ExternalGitRepo]):
+        assert self.should_check(external_data)
+
+        component_name = external_data.checker_data.get(
+            "component", ChromiumComponent.NAME
+        )
+        if component_name not in self._COMPONENTS:
+            raise ValueError(f"Invalid component {component_name}")
```

Comment:
I guess this is redundant, too: we'll get a `KeyError` from the next statement anyway if invalid component specified.

Reply:
It sort of is, but I figured this would be a bit clearer / require less knowledge of Python to understand.

Reply:
Well, I assume there won't be other users of this checker besides Chromium flatpak maintainers.